reconcile/vault_replication.py

```diff
@@ -20,33 +20,36 @@
 QONTRACT_INTEGRATION = "vault-replication"
 
 
+def deep_copy_versions(
+    dry_run: bool, source_vault, dest_vault, initial_version, last_version, path
+):
+    for version in range(initial_version, last_version + 1):
+        secret_dict = {"path": path, "version": version}
+        secret, version = source_vault.read_all_version(secret_dict)
```

Comment:
initial_version is the current version in dest, right? i think your index range starts 1 too soon

so lets say dest is at version 3 and source at version 5
initial_version=3, last_version=5 so you loop over [3,4,5]
you read version 3 from source and write it to dest, but it will become version 4 in dest because 3 exists already

Reply:
Changed the loop variables to clarify what was the actual values, and also added the +1 to match the expected outcome of copying all the versions

Thanks!